Src/ILGPU/Frontend/Intrinsic/LanguageIntrinsics.cs

```diff
@@ -0,0 +1,238 @@
+
+        /// <summary>
+        /// Returns the assigned intrinsic kind.
+        /// </summary>
+        public LanguageIntrinsicKind IntrinsicKind { get; }
+    }
+
+    partial class Intrinsics
+    {
+        /// <summary>
+        /// Handles language operations.
+        /// </summary>
+        /// <param name="context">The current invocation context.</param>
+        /// <param name="attribute">The intrinsic attribute.</param>
+        /// <returns>The resulting value.</returns>
+        private static ValueReference HandleLanguageOperation(
+            ref InvocationContext context,
+            LanguageIntrinsicAttribute attribute)
+        {
+            var builder = context.Builder;
+            return attribute.IntrinsicKind switch
+            {
+                LanguageIntrinsicKind.Version =>
+                    builder.CreateLanguageVersion(context.Location),
+                LanguageIntrinsicKind.EmitPTX =>
+                    CreateLanguageEmitPTX(ref context),
+                _ => throw context.Location.GetNotSupportedException(
+                    ErrorMessages.NotSupportedLanguageIntrinsic,
+                    attribute.IntrinsicKind.ToString()),
+            };
+        }
+
+        /// <summary>
+        /// Creates a new inline PTX instruction.
+        /// </summary>
+        /// <param name="ptxExpression">The PTX expression string.</param>
+        /// <param name="context">The current invocation context.</param>
+        private static ValueReference CreateLanguageEmitPTX(
+            string ptxExpression,
+            ref InvocationContext context)
+        {
+            // Parse PTX expression and ensure valid argument references
+            var location = context.Location;
+            if (!TryParse(ptxExpression, out var expressions))
+            {
+                throw location.GetNotSupportedException(
+                    ErrorMessages.NotSupportedInlinePTXFormat,
+                    ptxExpression);
+            }
+
+            // Validate all expressions
+            foreach (var expression in expressions)
+            {
+                if (!expression.HasArgument)
+                    continue;
+                if (expression.Argument < 0 ||
+                    expression.Argument >= context.NumArguments - 1)
+                {
+                    throw location.GetNotSupportedException(
+                        ErrorMessages.NotSupportedInlinePTXFormatArgumentRef,
+                        ptxExpression,
+                        expression.Argument);
+                }
+            }
+
+            // Gather all arguments
+            var arguments = InlineList<ValueReference>.Empty;
+            context.Arguments.CopyTo(ref arguments);
+            arguments.RemoveAt(0);
+
+            // Valid all argument types
+            foreach (var arg in arguments)
+            {
+                if (arg.BasicValueType == BasicValueType.Int16 ||
```

Comment:
May we add a comment here why `Int8` is not supported, since it cannot be represented properly in a non-existing 8-byte register?

Reply:
This restriction is because the [inline PTX assembly](https://docs.nvidia.com/cuda/inline-ptx-assembly/index.html) did not have a register constraint for smaller types. However, on second thought, I think we should allow all primitive types and just deal with it in the PTX emitter.